File: CS/XtraTabWithButtons/Tab/XtraTabControlDescendant.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraTab;

namespace XtraTabWithButtons
{
    public class XtraTabControlDescendant : XtraTabControl
    {
        protected override XtraTabPageCollection CreateTabCollection()
        {
            return new XtraTabPageCollectionDescendant(this);
        }
    }
}

File: VB/XtraTabWithButtons/Tab/XtraTabControlDescendant.vb
Imports DevExpress.XtraTab

Namespace XtraTabWithButtons

    Public Class XtraTabControlDescendant
        Inherits XtraTabControl

        Protected Overrides Function CreateTabCollection() As XtraTabPageCollection
            Return New XtraTabPageCollectionDescendant(Me)
        End Function
    End Class
End Namespace
